apps/wakunode2/wakunode2.nim

```diff
@@ -431,27 +490,42 @@ proc startNode(node: WakuNode, conf: WakuNodeConf,
   
   return ok()
 
-proc startExternal(node: WakuNode, conf: WakuNodeConf): SetupResult[void] =
-  ## Start configured external interfaces and monitoring tools
-  ## on a Waku v2 node, including the RPC API, REST API and metrics
-  ## monitoring ports.
+proc resumeMessageStore(node: WakuNode, address: string): Future[SetupResult[void]] {.async.} =
+  # Resume historical messages, this has to be called after the node has been started
+  if address != "":
+    return err("empty peer multiaddres")
   
-  if conf.rpc:
-    try:
-      startRpcServer(node, conf.rpcAddress, Port(conf.rpcPort + conf.portsShift), conf)
-    except:
-      return err("failed to start the json-rpc server: " & getCurrentExceptionMsg())
+  var remotePeer: RemotePeerInfo
+  try:
+    remotePeer = parseRemotePeerInfo(address)
+  except:
+    return err("invalid peer multiaddress: " & getCurrentExceptionMsg())
   
-  if conf.rest:
-    startRestServer(node, conf.restAddress, Port(conf.restPort + conf.portsShift), conf)
+  try:
+    await node.resume(some(@[remotePeer]))
+  except:
+    return err("failed to resume messages history: " & getCurrentExceptionMsg())
 
-  if conf.metricsLogging:
-    startMetricsLog()
 
-  if conf.metricsServer:
-    startMetricsServer(conf.metricsServerAddress,
-      Port(conf.metricsServerPort + conf.portsShift))
+proc startRpcServer(node: WakuNode, address: ValidIpAddress, port: uint16, portsShift: uint16, conf: WakuNodeConf): SetupResult[void] =
+  try:
+    startRpcServer(node, address, Port(port + portsShift), conf)
+  except:
+    return err("failed to start the json-rpc server: " & getCurrentExceptionMsg())
+
+  ok()
   
+proc startRestServer(node: WakuNode, address: ValidIpAddress, port: uint16, portsShift: uint16, conf: WakuNodeConf): SetupResult[void] =
+  startRestServer(node, address, Port(port + portsShift), conf)
+  ok()
+
+proc startMetricsServer(node: WakuNode, address: ValidIpAddress, port: uint16, portsShift: uint16): SetupResult[void] =
+  startMetricsServer(address, Port(port + portsShift))
```

Comment:
a try-except here maybe?

Reply:
Well... the compiler is not complaining, so I don't think we should add a try/catch there.

But I agree that the error management in the metrics, JSON-RPC, and rest servers setup procedure has to be improved.